java/src/main/java/org/rocksdb/WriteBatch.java

```diff
@@ -59,8 +59,21 @@ public void iterate(final Handler handler) throws RocksDBException {
    * @param nativeHandle address of native instance.
    */
   WriteBatch(final long nativeHandle) {
+    this(nativeHandle, true);
+  }
+
+  /**
+   * <p>Private WriteBatch constructor which is used to construct
+   * WriteBatch instances from C++ side. As the reference to this
+   * object is also managed from C++ side the handle will be disowned.</p>
+   *
+   * @param nativeHandle address of native instance.
+   * @param disOwn whether to own this reference from the C++ side or not
+   */
+  WriteBatch(final long nativeHandle, boolean disOwn) {
```

Comment:
I would like to see the `disOwn` argument inverted, so that it is consistent with `RocksMutableObject` etc. Could you change the argument to `final boolean owningNativeHandle`.

Also the javadoc for this additional constructor does not make sense, as this method is not called from C++ but is called from Java.